Support 'display: masonry' so we can run WPT tests [css-grid-3]

fantasai opened the pull request:
#### ec8a41f79f5b4bcbb801f3abb5f2f814db410d61
<pre>
Support &apos;display: masonry&apos; so we can run WPT tests [css-grid-3]
<a href="https://bugs.webkit.org/show_bug.cgi?id=301907">https://bugs.webkit.org/show_bug.cgi?id=301907</a>
<a href="https://rdar.apple.com/163985106">rdar://163985106</a>

Reviewed by Brandon Stewart, Tim Nguyen, and Simon Fraser.

The CSSWG resolved to use &apos;display&apos; for switching into masonry-style layout,
but hasn&apos;t decided on a keyword yet. However all the WPTs are using
&apos;display: masonry&apos;, which means we need to support this syntax if we want
to run the WPTs. We will update to the final keyword later.

This patch removes &apos;masonry&apos; from the &apos;grid-template&apos; properties,
adds support for &apos;masonry&apos; and &apos;inline-masonry&apos; to the &apos;display&apos; property,
and re-uses the &apos;grid-auto-flow&apos; property for orientation control.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-display/parsing/tentative/display-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-display/parsing/tentative/display-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-automatic-minimum-for-auto-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-fit-content-percentage-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-minimum-contribution-with-percentages-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-minimum-size-grid-items-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-minimum-size-grid-items-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-automatic-minimum-for-auto-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-minimum-contribution-with-percentages-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-minimum-size-grid-items-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-minimum-size-grid-items-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-009-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-010-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-011-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-012-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-019-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-020-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-024-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-025-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-015-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-016-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-020-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-021-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-015-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-016-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-020-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-021-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-007-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-010-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-011-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-017-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-018-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-022-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-023-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/dynamic-grid-track-direction-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-009-expected.txt: Added.

Rebaseline tests.

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/alignment/column-justify-items-end-justify-self-start-001.html:
* LayoutTests/platform/gtk/TestExpectations:

Linux-specific failures.

* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::displayTypeNeedsSpace):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Display.cpp:
(WebCore::CSSPropertyParserHelpers::consumeDisplay):
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Grid.cpp:
(WebCore::CSSPropertyParserHelpers::consumeGridTrackList):
(WebCore::CSSPropertyParserHelpers::consumeGridTemplatesRowsOrColumns):
* Source/WebCore/html/HTMLButtonElement.cpp:
(WebCore::HTMLButtonElement::createElementRenderer):
* Source/WebCore/layout/integration/grid/LayoutIntegrationGridCoverage.cpp:
(WebCore::LayoutIntegration::gridLayoutAvoidanceReason):
* Source/WebCore/layout/layouttree/LayoutBox.cpp:
(WebCore::Layout::Box::isBlockLevelBox const):
(WebCore::Layout::Box::isInlineLevelBox const):
* Source/WebCore/layout/layouttree/LayoutBox.h:
(WebCore::Layout::Box::isGridBox const):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::createFor):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::isMasonry const):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::isDisplayBlockType):
(WebCore::RenderStyle::isDisplayInlineType):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::equivalentBlockDisplay):
(WebCore::Style::equivalentInlineDisplay):
(WebCore::Style::shouldInheritTextDecorationsInEffect):
* Source/WebCore/style/StyleExtractorCustom.h:
(WebCore::Style::extractGridTemplateValue):
* Source/WebCore/style/values/grid/StyleGridTemplateList.cpp:
(WebCore::Style::CSSValueConversion&lt;GridTemplateList&gt;::operator):
(WebCore::Style::Blending&lt;GridTemplateList&gt;::canBlend):
(WebCore::Style::Blending&lt;GridTemplateList&gt;::blend):
(WebCore::Style::operator&lt;&lt;):
* Source/WebCore/style/values/grid/StyleGridTemplateList.h:

Canonical link: <a href="https://commits.webkit.org/302746@main">https://commits.webkit.org/302746@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ce4d89450555bc89c1cf43a16b0cc5bfaae59763

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130028 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2289 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/40889 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137423 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/81540 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/ad1eeb9f-a38d-46f5-a3bc-d6bfb3f58ed7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/131899 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/2254 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2181 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99050 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/66848 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/68eb65d9-95f9-4c08-9382-0a066355e228) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132975 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1676 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/116465 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/79750 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/1595 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/34590 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/80695 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110120 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35100 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/139902 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2083 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1931 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/107558 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/2128 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/112812 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/107451 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27357 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/1664 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31264 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/54916 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/2156 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/65525 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1972 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/2005 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2079 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Display.cpp:
```diff
@@ -42,19 +42,20 @@ RefPtr<CSSValue> consumeDisplay(CSSParserTokenRange& range, CSS::PropertyParserS
 
     // Parse single keyword values
-    auto singleKeyword = consumeIdent<
+    auto singleKeyword = state.context.itemPackCollapseDisplayGridEnabled && (range.peek().id() == CSSValueMasonry || range.peek().id() == CSSValueInlineMasonry) ? consumeIdent(range) : consumeIdent<
```
This line is trying to do a significant amount of work.

1. Checking if `display: masonry` feature is enabled.
2. Peeking for the next id and checking if it Masonry or an inline Masonry.
3. Finally consuming the ident.

Can we refactor this.

stwrt commented on Source/WebCore/rendering/style/RenderStyleConstants.cpp:
```diff
@@ -483,6 +483,8 @@ TextStream& operator<<(TextStream& ts, DisplayType display)
     case DisplayType::Contents: ts << "contents"_s; break;
     case DisplayType::Grid: ts << "grid"_s; break;
     case DisplayType::InlineGrid: ts << "inline-grid"_s; break;
+    case DisplayType::Masonry: ts << "grid"_s; break;
```
Should this be `masonry` and `inline-masonry`?

fantasai replied:
Yeah, good point, I'll fix it.

stwrt commented on Source/WebCore/layout/integration/grid/LayoutIntegrationGridCoverage.cpp:
```diff
@@ -72,6 +72,7 @@ enum class AvoidanceReason : uint64_t {
     GridItemHasContainsSize = 1LLU << 35,
     GridItemHasUnsupportedColumnPlacement = 1LLU << 36,
     GridItemHasUnsupportedRowPlacement = 1LLU << 37,
+    NotAGrid = 1LLU << 38,
```
This would only be a Masonry or inline Masonry right? I would mention that instead of just not a grid.

fantasai replied:
Probably, but I don't know what else we might be feeding through this code path--the check is just for `grid`/`inline-grid` display types.

nt1m replied:
We think masonry _is_ a grid :)

stwrt commented on LayoutTests/TestExpectations:
```diff
@@ -1882,33 +1851,25 @@ imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/subgrid/masonry-s
-imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-005.html [ ImageOnlyFailure ]
 imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-006.html [ ImageOnlyFailure ]
 imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-007.html [ ImageOnlyFailure ]
+imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-008.html [ ImageOnlyFailure ]
```
Why are we failing 4 new tests?

Glad we are passing 45 new tests 😊.

fantasai replied:
Probably because the reference was written in a way that if you don't support masonry at all, the test and the reference happen to match...

nt1m commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Display.cpp:
```diff
@@ -162,6 +168,8 @@ RefPtr<CSSValue> consumeDisplay(CSSParserTokenRange& range, CSS::PropertyParserS
             return CSSValueInlineBlock;
         case CSSValueGrid:
             return CSSValueInlineGrid;
+        case CSSValueMasonry:
+            return CSSValueInlineMasonry;
```
Do we need to convert `inline masonry` to `inline-masonry` ? We do it for the other ones for compat purposes, but not sure it's valuable for `inline-masonry`. I guess it might needed for the value representation?

fantasai replied:
We do, unless I refactor how we store display values to be a multi-valued struct instead of an enum. :/ I thought about it, but figured that was better to tackle as a follow-up sometime later.

nt1m commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Display.cpp:
```diff
@@ -122,6 +127,7 @@ RefPtr<CSSValue> consumeDisplay(CSSParserTokenRange& range, CSS::PropertyParserS
         case CSSValueFlow:
         case CSSValueFlowRoot:
         case CSSValueGrid:
+        case CSSValueMasonry:
```
This is not gated behind a preference.

fantasai replied:
Thanks for catching this. I was expecting I would miss something here, and figured you'd catch it. :P

nt1m commented on Source/WebCore/css/parser/CSSPropertyParserConsumer+Display.cpp:
```diff
@@ -42,44 +42,49 @@ RefPtr<CSSValue> consumeDisplay(CSSParserTokenRange& range, CSS::PropertyParserS
-    >(range);
+    auto singleKeyword = [&]() {
+        if (state.context.itemPackCollapseDisplayGridEnabled && (range.peek().id() == CSSValueMasonry || range.peek().id() == CSSValueInlineMasonry))
+            return consumeIdent(range);
```
singleKeyword is only for keywords that exclusively single.

fantasai replied:
Fixed, thanks!

nt1m commented on Source/WebCore/css/CSSValueKeywords.in:
```diff
@@ -531,6 +531,8 @@ inline-flex
 grid
 inline-grid
+masonry
+inline-masonry
```
Is that needed? I thought the dashed keywords were only for compatibility, e.g. `display: inline masonry` being the correct way to do this.

fantasai replied:
Yeah, I agree, but I think that's a CSSWG issue. My goal here is to get all the tests to run so we can check layout.

Loirooriol replied:
Discussion in https://github.com/w3c/csswg-drafts/issues/10961

nt1m commented on Source/WebCore/rendering/RenderGrid.cpp:
```diff
@@ -1102,7 +1102,9 @@ bool RenderGrid::isMasonry(Style::GridTrackSizingDirection direction) const
-    return tracks.masonry;
+    if (style().display() != DisplayType::Masonry && style().display() != DisplayType::InlineMasonry)
+        return false;
+    return (direction == Style::GridTrackSizingDirection::Columns) == style().gridAutoFlow().isColumn();
```
Do we not support masonry in both axises?

fantasai replied:
Yes, but this function is asking "is this axis masonry", and we need to check `grid-auto-flow` to determine which axis is masonry.

nt1m replied:
The naming of this function doesn't make that very clear. Maybe a possible followup rename?

fantasai replied:
I think the function is clear if you consider the parameter it's taking. You pass in an axis, and it's asking if that axis is masonry.

The thing that bugs me here is having `isMasonry(...::Rows)` and `areMasonryRows()` be inconsistent. :p

stwrt reviewed (commented):
> Do we not support masonry in both axises? 

We should support masonry in both axises.

fantasai committed:
Support 'display: masonry' so we can run WPT tests [css-grid-3]
https://bugs.webkit.org/show_bug.cgi?id=301907
rdar://163985106

Reviewed by Brandon Stewart, Tim Nguyen, and Simon Fraser.

The CSSWG resolved to use 'display' for switching into masonry-style layout,
but hasn't decided on a keyword yet. However all the WPTs are using
'display: masonry', which means we need to support this syntax if we want
to run the WPTs. We will update to the final keyword later.

This patch removes 'masonry' from the 'grid-template' properties,
adds support for 'masonry' and 'inline-masonry' to the 'display' property,
and re-uses the 'grid-auto-flow' property for orientation control.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-display/parsing/tentative/display-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-display/parsing/tentative/display-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-automatic-minimum-for-auto-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-fit-content-percentage-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-minimum-contribution-with-percentages-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-minimum-size-grid-items-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/column-minimum-size-grid-items-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-automatic-minimum-for-auto-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-minimum-contribution-with-percentages-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-minimum-size-grid-items-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/items/row-minimum-size-grid-items-003-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-009-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-010-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-011-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-012-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-019-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-020-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-024-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/column-auto-repeat-025-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-015-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-016-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-020-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/column-auto-repeat-auto-021-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-015-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-016-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-020-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/intrinsic-auto-repeat/row-auto-repeat-auto-021-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-007-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-008-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-010-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-011-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-017-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-018-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-022-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-023-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/dynamic-grid-track-direction-expected.txt:
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/track-sizing/auto-repeat/row-auto-repeat-009-expected.txt: Added.

Rebaseline tests.

* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/masonry/tentative/alignment/column-justify-items-end-justify-self-start-001.html:
* LayoutTests/platform/gtk/TestExpectations:

Linux-specific failures.

* Source/WebCore/accessibility/AccessibilityNodeObject.cpp:
(WebCore::displayTypeNeedsSpace):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
(WebCore::toCSSValueID):
(WebCore::fromCSSValueID):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/CSSValueKeywords.in:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Display.cpp:
(WebCore::CSSPropertyParserHelpers::consumeDisplay):
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Grid.cpp:
(WebCore::CSSPropertyParserHelpers::consumeGridTrackList):
(WebCore::CSSPropertyParserHelpers::consumeGridTemplatesRowsOrColumns):
* Source/WebCore/html/HTMLButtonElement.cpp:
(WebCore::HTMLButtonElement::createElementRenderer):
* Source/WebCore/layout/integration/grid/LayoutIntegrationGridCoverage.cpp:
(WebCore::LayoutIntegration::gridLayoutAvoidanceReason):
* Source/WebCore/layout/layouttree/LayoutBox.cpp:
(WebCore::Layout::Box::isBlockLevelBox const):
(WebCore::Layout::Box::isInlineLevelBox const):
* Source/WebCore/layout/layouttree/LayoutBox.h:
(WebCore::Layout::Box::isGridBox const):
* Source/WebCore/rendering/RenderElement.cpp:
(WebCore::RenderElement::createFor):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::isMasonry const):
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator<<):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
(WebCore::RenderStyle::isDisplayBlockType):
(WebCore::RenderStyle::isDisplayInlineType):
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::equivalentBlockDisplay):
(WebCore::Style::equivalentInlineDisplay):
(WebCore::Style::shouldInheritTextDecorationsInEffect):
* Source/WebCore/style/StyleExtractorCustom.h:
(WebCore::Style::extractGridTemplateValue):
* Source/WebCore/style/values/grid/StyleGridTemplateList.cpp:
(WebCore::Style::CSSValueConversion<GridTemplateList>::operator):
(WebCore::Style::Blending<GridTemplateList>::canBlend):
(WebCore::Style::Blending<GridTemplateList>::blend):
(WebCore::Style::operator<<):
* Source/WebCore/style/values/grid/StyleGridTemplateList.h:

Canonical link: https://commits.webkit.org/302746@main
